src/components/Dropdown/Dropdown.types.ts

```diff
@@ -17,22 +17,28 @@ export type DropdownDataOption = {
     attributes?: React.HTMLAttributes<HTMLElement>;
     image?: React.ReactElement;
     imageSize?: DropdownImageSize;
-    groupLabel?: string;
-    options?: [DropdownDataOption];
 }
 
+export type DropdownDataGrouped = {
+    groupLabel: string;
+    options: DropdownDataOption[];
+}
+
+export type DropdownData = DropdownDataOption[] | DropdownDataGrouped[];
```

Comment:
If the two things can be interleaved, it should be `Array<DropdownDataOption | DropdownDataGrouped>`

Otherwise, it's good as it is right now

Reply:
It's exclusive `DropdownData` can be `DropdownDataOption[]` OR `DropdownDataGrouped[]`